unlock-app/src/middlewares/wedlocksMiddleware.ts

```diff
@@ -10,8 +10,12 @@ const wedlocksMiddleware = (config: any) => {
     return (next: any) => {
       return (action: Action) => {
         if (action.type === SIGNUP_EMAIL) {
+          const { origin } = window.location
```

Comment:
window may not be available on server-side, you should use `global.location`

Reply:
Typescript doesn't like referencing `global.location`. I think that there's no way for the `SIGNUP_EMAIL` action to be produced server-side since it's only emitted as a result of a user interacting with the browser.

If it's important I can look into what kind of type patching I'll need to do to make it work with global, but I think this is a case where use of `window` shouldn't be a problem.

Reply:
I think that because next aliases global to window, perhaps they have figured out the typing issues?

Reply:
Looks like the solution is to patch the `global` type. Rather than doing that, which may hide errors in the future (if we change frameworks or something), why don't we add a check that window is actually available?

Reply:
sounds like a good solution